orc8r/gateway/python/magma/magmad/config_manager.py

```diff
@@ -73,6 +74,34 @@ def get_request_args(self, stream_name: str) -> Any:
         )
         return mconfig_digest_proto
 
+    def _warn_if_versions_are_incompatible(self, agw_version, unpacked_mconfig):
+        # version should be in X.X.X format with only non-negative numbers allowed for X
+        VERSION_REGEX = re.compile(r"[0-9]+\.(?P<minor_version>[0-9]+)\.[0-9]+")
+
+        # unpack the magmad structure to get orce_version field
+        if unpacked_mconfig.Is(mconfigs_pb2.MagmaD.DESCRIPTOR):
+            magmad_parsed = mconfigs_pb2.MagmaD()
+            unpacked_mconfig.Unpack(magmad_parsed)
+            orc8r_version = magmad_parsed.orc8r_version
+
+            agw_version_parsed = VERSION_REGEX.match(agw_version)
+            orc8r_version_parsed = VERSION_REGEX.match(orc8r_version)
+
+            # agw_version is not in expected format
+            if not agw_version_parsed:
+                logging.warning("Gateway version: %s not valid" % agw_version)
+
+            # orc8r_version is not in expected format
+            if not orc8r_version_parsed:
+                logging.warning("Orchestrator version: %s not valid" % orc8r_version)
+
+            return (agw_version_parsed, orc8r_version_parsed)
```

Comment:
I would recommend to define a new `NamedTuple` with both properties and return that, which also helps with adding type hints